src/components/CommentForm.jsx

```diff
@@ -1,9 +1,36 @@
-import { convertDateLong, convertDateShort } from "../utils";
+import { useState } from "react";
+import { convertDateLong, pulseMsg } from "../utils";
 import { Card } from "./Card";
+import { postComment } from "../client";
+
+export function CommentForm({ article_id, setComments }) {
+  const [commentInput, setCommentInput] = useState("");
+  const [commentMsg, setCommentMsg] = useState(null);
+  const [postedNewComment, setPostedNewComment] = useState(true);
 
-export function CommentForm({ article_id }) {
   function handleSubmit(event) {
     event.preventDefault();
+    if (commentInput.length > 0) {
+      pulseMsg("Posting...", setCommentMsg);
+      postComment({
+        article_id,
+        username: "grumpy19",
+        body: commentInput,
+      })
+        .then(({ comment }) => {
+          setCommentInput("");
+          setComments((comments) => [comment, ...comments]);
```

Comment:
I like that you've put the comment first, really nice to see it appear underneath the text box